CMakeLists.txt

```diff
@@ -624,30 +622,44 @@ if(AVIF_BUILD_APPS OR (AVIF_BUILD_TESTS AND (AVIF_ENABLE_FUZZTEST OR AVIF_ENABLE
     endif()
 
     add_library(
-        avif_apps STATIC apps/shared/avifexif.c apps/shared/avifjpeg.c apps/shared/avifpng.c apps/shared/avifutil.c
-                         apps/shared/iccmaker.c apps/shared/y4m.c third_party/iccjpeg/iccjpeg.c
+        avif_apps_obj OBJECT $<TARGET_OBJECTS:avif_obj> apps/shared/avifexif.c apps/shared/avifjpeg.c apps/shared/avifpng.c
+                             apps/shared/avifutil.c apps/shared/iccmaker.c apps/shared/y4m.c third_party/iccjpeg/iccjpeg.c
     )
-    target_link_libraries(avif_apps avif PNG::PNG ZLIB::ZLIB JPEG::JPEG)
+    set_property(TARGET avif_apps_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
This change does not seem necessary.

If this is necessary or desirable, then we should also compile avifdec.c and avifenc.c as position-independent code, right?

Like my comment at line 515 above, if this change can stand alone, it is better to make this change in a separate pull request.